### PR TITLE
Collection loading opts

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: engine/gameobject/src/gameobject/gameobject.cpp [hunk 1]
```diff
@@ -287,9 +287,13 @@ namespace dmGameObject
 
     void DeallocCollection(Collection* collection)
     {
+        DM_PROFILE(GameObject, "DeallocCollection");
+
         HRegister regist = collection->m_Register;
         for (uint32_t i = 0; i < regist->m_ComponentTypeCount; ++i)
         {
+            DM_PROFILE_DYN(GameObjectDeleteWorld, regist->m_ComponentTypes[i].m_Name, regist->m_ComponentTypes[i].m_NameHash);
```
Comment (JCash) [inline engine/gameobject/src/gameobject/gameobject.cpp:295]: Make it visible that the deallocation for different component types actually take time.

File: engine/graphics/src/opengl/graphics_opengl.cpp [hunk 2]
```diff
@@ -2026,17 +2026,44 @@ static uintptr_t GetExtProcAddress(const char* name, const char* extension_name,
...
+        JobDesc j;
+        j.m_Context = (void*)texture;
+        j.m_Func = OpenGLDoDeleteTexture;
+        j.m_FuncComplete = 0;
+        JobQueuePush(j);
+    }
+
     static void PostDeleteTextures(bool force_delete)
     {
+        DM_PROFILE(Graphics, "PostDeleteTextures");
```
Comment (JCash) [inline engine/graphics/src/opengl/graphics_opengl.cpp:2048]: This function is called at the end of each frame.

File: engine/graphics/src/opengl/graphics_opengl.cpp [hunk 3]
```diff
@@ -2026,17 +2026,44 @@ static uintptr_t GetExtProcAddress(const char* name, const char* extension_name,
...
+
+        if (force_delete)
+        {
+            uint32_t size = g_PostDeleteTexturesArray.Size();
+            for (uint32_t i = 0; i < size; ++i)
+            {
+                OpenGLDoDeleteTexture(g_PostDeleteTexturesArray[i]);
```
Comment (JCash) [inline engine/graphics/src/opengl/graphics_opengl.cpp:2055]: If we're shutting down, delete the textures right away.

File: engine/graphics/src/opengl/graphics_opengl.cpp [hunk 4]
```diff
@@ -2026,17 +2026,44 @@ static uintptr_t GetExtProcAddress(const char* name, const char* extension_name,
...
         while(i < g_PostDeleteTexturesArray.Size())
         {
             HTexture texture = g_PostDeleteTexturesArray[i];
-            if((!(dmGraphics::GetTextureStatusFlags(texture) & dmGraphics::TEXTURE_STATUS_DATA_PENDING)) || (force_delete))
+            if(!(dmGraphics::GetTextureStatusFlags(texture) & dmGraphics::TEXTURE_STATUS_DATA_PENDING))
             {
-                glDeleteTextures(1, &texture->m_Texture);
-                CHECK_GL_ERROR;
-                delete texture;
+                OpenGLDeleteTextureAsync(texture);
```
Comment (JCash) [inline engine/graphics/src/opengl/graphics_opengl.cpp:2066]: ..if not, put them on the async thread for deletion.

File: engine/gui/src/gui.cpp [hunk 5]
```diff
@@ -3646,22 +3659,19 @@ Result DeleteDynamicTexture(HScene scene, const dmhash_t texture_hash)
         Animation animation;
         uint32_t animation_index = 0xffffffff;
 
-        // Remove old animation for the same property
-        for (uint32_t i = 0; i < scene->m_Animations.Size(); ++i)
+        uint32_t* animation_index_ptr = scene->m_ValueToAnimationIndex.Get(PtrToUint(value));
```
Comment (JCash) [inline engine/gui/src/gui.cpp]: Using a hashtable was a huge time saver.
Going from ~150ms, to ~95ms in a very large scene.

File: engine/gamesys/src/gamesys/components/comp_sound.cpp [hunk 6]
```diff
@@ -90,18 +90,6 @@ namespace dmGameSystem
             if (entry.m_SoundInstance != 0)
             {
                 dmSound::Stop(entry.m_SoundInstance);
-            }
-        }
-
-        // An update is required after dmSound::Stop and the follwing dmSound::DeleteSoundInstance
```
Comment (britzl) [inline engine/gamesys/src/gamesys/components/comp_sound.cpp:96]: Is this assumption no longer true?

File: engine/gui/src/gui.cpp [hunk 7]
```diff
@@ -1493,6 +1494,14 @@ Result DeleteDynamicTexture(HScene scene, const dmhash_t texture_hash)
         }
     }
 
+    // Removes the last zero bits, allowing for better distribution in a hash table
```
Comment (britzl) [inline engine/gui/src/gui.cpp:1497]: Interesting!

**Review by britzl — DISMISSED** [review]
Looks good!

**Pull request merged**